### PR TITLE
fix: Fix URL Checker

### LINKED ISSUE #3386: Fix the URL checker
lychee is failing here at the moment -

https://github.com/pybamm-team/PyBaMM/blob/7eb5879c264e26b02a71b0ced8ef11a5f092e451/CONTRIBUTING.md?plain=1#L188

We could use a different material to reference glob pattern?

### PR DESCRIPTION
**Pull request by Agnik7**
# Description

Fixed the broken glob pattern link in CONTRIBUTING.md. I have replaced the previous link with the following [link](https://docs.python.org/3/library/glob.html)
Fixes #3386 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by Agnik7** [comment]
@Saransh-cpp I have created the PR. Requesting for your review.

**Comment by agriyakhetarpal** [comment]
Thanks, @Agnik7! We can skip adding this in the CHANGELOG.

**Comment by Agnik7** [comment]
Ok, I'm removing it from the CHANGELOG

**Comment by Agnik7** [comment]
@agriyakhetarpal I have removed it. Please check.

**Comment by Agnik7** [comment]
A request to the maintainers of this repository: I am participating in Hacktoberfest, so if you merge this PR, kindly add the hacktoberfest and hacktoberfest-accepted label to it. Apologies for any inconvenience caused.

File: CONTRIBUTING.md
Comment (kratman) [inline CONTRIBUTING.md:188]: This seems like a link to another software package. Is there something more general that could be used? Like Linux documentation or python documentation?

Comment (Agnik7) [reply CONTRIBUTING.md]: @kratman I have changed the link to a Linux documentation link. Is it alright? Please check

**Review by arjxn-py — COMMENTED** [review]
Thanks @Agnik7, I can see that the linkChecker is still failing. 
Can you please look into it? Maybe we can try with [official python documentation](https://github.com/pybamm-team/PyBaMM/pull/3393/files)

**Comment by Agnik7** [comment]
@arjxn-py Alright changing that right now.

**Comment by Agnik7** [comment]
@arjxn-py I have changed the link. Please check.

**Review by arjxn-py — APPROVED** [review]
Thanks 🙂, looks good to me.

**Review by brosaplanella — APPROVED** [review]
Looks good to me, thanks!

**Comment by brosaplanella** [comment]
@all-contributors please add @Agnik7 for documentation

**Comment by brosaplanella** [comment]
@Agnik7, going forward it is better to create a new branch in your fork rather than work on the develop branch, as this can cause issues.

**Comment by Agnik7** [comment]
@brosaplanella alright! I'll surely keep that in mind

**Comment by brosaplanella** [comment]
Happy to merge once @kratman has checked the new changes :)

**Comment by Agnik7** [comment]
Thanks a lot @brosaplanella for your guidance!!! Requesting @kratman to review this PR. Apologies for any inconvenience caused

**Review by kratman — APPROVED** [review]
Sorry, did not realize you guys were waiting on me.

This is one of the links I found when I was reviewing this initially and I think it is a reasonable one. The only other link I saw that seemed reputable was from a unix man pages project (man7.org), but python documentation is probably better for now.

**Comment by Agnik7** [comment]
Thanks a lot @brosaplanella and @kratman for your invaluable suggestions and guidance throughout!! Requesting @brosaplanella a follow up to this PR. Thanks a lot!!!

**Comment by Agnik7** [comment]
@brosaplanella I have upstreamed this PR. Kindly approve the checks. Apologies for any inconvenience caused.

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Sorry for taking too long to approve the tests. There are a couple of more failing URLs, could you please fix them in this PR itself?

1. https://wigging.me should be changed to https://gavinw.me in `README.md` and here -

https://github.com/pybamm-team/PyBaMM/blob/d02a78b6edb262ee4a582b35f4c11eba22e08fd8/.all-contributorsrc#L445

2. The hackerearth blog post should be replaced with any alternative blog post discussing Python for R users.

**Comment by Agnik7** [comment]
@Saransh-cpp alright, I'm on it!!

**Comment by Agnik7** [comment]
@Saransh-cpp I have made the changes. Please have a look at them.

**Review by Saransh-cpp — APPROVED** [review]
Looks good, thanks, @Agnik7!

**Pull request merged**
